### PR TITLE
feat: [sc-25514] split NameGuard code into separate package / app

### PR DESCRIPTION
**Pull request by Carbon225**
Story details: https://app.shortcut.com/ps-web3/story/25514

### CODE REVIEW

**Comment by Carbon225** [comment]
I did this:

```yaml
provider:
  name: aws
  stage: ${self:custom.stage}
  architecture: arm64
  ecr:
    images:
      oss-nameguard:
        # /packages/nameguard-python/
        path: ../../packages/nameguard-python/
      platform: linux/arm64
```

The dockerfile is in nameguard-python and serverless.yaml refers to it by relative paths.

**Comment by lightwalker-eth** [comment]
@Carbon225 Replying to your question:

> @djstrong @lightwalker-eth do you think we can use the PyPI published nameguard package for the lambda? Or does it have to be built from the repository directly? In the old lambda dockerfile we installed NG from the current directory but now I changed it to install NG from pip. That's because I'm not sure how to correctly refer to packages/nameguard-python from apps/api.nameguard.io/serverless.yaml or from the dockerfile.

Good question. At first my thought was to like the idea of using the PyPI published NameGuard package when we build and deploy the lambda. In this thinking, the "app" for "api.nameguard.io" should be possible to build and deploy independently with reference to the PyPI dependency, without needing to create a local build of the NameGuard package.

But I have a few concerns / questions with this approach:

1. I'm worried about our CI. This would mean that each time we want to update the NameGuard Lambda deployment we would need to ensure that we completed the step to update the PyPI published package first. I'm not immediately sure how we would achieve this unless we merged both of these GitHub Actions into a single script?
2. ... additionally, small edge case, but not so sure about potential race conditions between publishing packages updates to PyPI and building and deploying the NameGuard Lambda.
3. ... additionally, we have an important goal that's about to be prioritized in our roadmap to create a dedicated "staging" environment for the NameGuard Lambda. Ideally there would be minimal differences between how we deploy to staging vs. production. So then we would need to create a special PyPI distribution for a "staging" distribution of the NameGuard Lambda. And not sure that we want to be doing that and making each staging distribution public on PyPI in this way?

It all starts looking pretty complicated..

Maybe best to make it so that the "app" for "api.nameguard.io" always performs its own build of NameGuard directly from the monorepo and not to use PyPI?

**Comment by lightwalker-eth** [comment]
@Carbon225 Replying to your comment:

> I did this:
> 
> ```yaml
> provider:
>   name: aws
>   stage: ${self:custom.stage}
>   architecture: arm64
>   ecr:
>     images:
>       oss-nameguard:
>         # /packages/nameguard-python/
>         path: ../../packages/nameguard-python/
>       platform: linux/arm64
> ```
> 
> The dockerfile is in nameguard-python and serverless.yaml refers to it by relative paths.

Did this work? If so, cool :+1:

**Comment by djstrong** [comment]
I have moved also `package.json` to `apps/api.nameguard.io`.
After deploying to staging, tests pass: `LAMBDA_ROOT_URL=https://api-staging.nameguard.io MONKEYPATCH=0 poetry run pytest`

**Comment by djstrong** [comment]
Lambda timeout has been increased to 60 seconds.

**Review by lightwalker-eth — CHANGES_REQUESTED** [review]
@Carbon225 @djstrong Great to see these updates. Reviewed and shared feedback. Also created a PR for you with additional suggestions: https://github.com/namehash/namekit/pull/442

**Comment by lightwalker-eth** [comment]
Please also check the following as I think they need updates too:

- `.pre-commit-config.yaml` (I assume this should now consider all directories containing python code?)

**Comment by djstrong** [comment]
> * `.pre-commit-config.yaml` (I assume this should now consider all directories containing python code?)

I think there is only Python directory `packages/nameguard-python`.

File: packages/nameguard-python/.docker/config.json
[the remark this answers is not in the document]
Comment (djstrong) [reply packages/nameguard-python/.docker/config.json:1]: I have created story for this: https://app.shortcut.com/ps-web3/story/25880/move-nameguard-s-docker-files-to-apps

**Review by lightwalker-eth — APPROVED** [review]
Looks good :+1:

**Pull request merged**